### PR TITLE
Use long SHAs in AzDO PR descriptions

### PR DESCRIPTION
**Pull request by dkurepa**
https://github.com/dotnet/arcade-services/issues/3114

I couldn't find any document online that says that using short shas works on AzDo, like it does on GitHub. Also did manual testing

### Release Note Category
- [ ] Feature changes/additions 
- [x] Bug fixes
- [ ] Internal Infrastructure Improvements

### Release Note Description
Use full sha for commit comparison on AzDo

### CODE REVIEW

File: src/Maestro/SubscriptionActorService/PullRequestDescriptionBuilder.cs [hunk 1]
```diff
@@ -176,28 +176,30 @@ public int GetStartingReferenceId()
...
     {
         if (repoURI == null)
         {
             throw new ArgumentNullException(nameof(repoURI));
         }
-        if (from == null)
+        if (fromSha == null)
         {
-            throw new ArgumentNullException(nameof(from));
+            throw new ArgumentNullException(nameof(fromSha));
```
Comment (premun) [inline src/Maestro/SubscriptionActorService/PullRequestDescriptionBuilder.cs:187]: Btw there is now this:
https://learn.microsoft.com/en-us/dotnet/api/system.argumentnullexception.throwifnull?view=net-8.0

```csharp
ArgumentNullException.ThrowIfNull(fromSha);
```

**Review by premun — COMMENTED** [review]
Great, had not idea we have this functionality. I have duplicated this in the VMR sync because it was hidden in this project (however, did not implement AzDO part yet). So I will probably move it to `DarcLib` eventually.

**Comment by dkurepa** [comment]
> Great, had not idea we have this functionality. I have duplicated this in the VMR sync because it was hidden in this project (however, did not implement AzDO part yet). So I will probably move it to `DarcLib` eventually.

I'm trying to figure out if it's the correct thing to do. AzDo has a limit of 4000 chars in the description and these automated ones can get quite big, for example https://dev.azure.com/dnceng/internal/_git/dotnet-installer/pullrequest/35851 has 3602, and that's the short sha that's not working. This change might tip it over and I'm not sure what happens then

File: src/Maestro/SubscriptionActorService/PullRequestDescriptionBuilder.cs (hunk 1 above)
Comment (dkurepa) [reply src/Maestro/SubscriptionActorService/PullRequestDescriptionBuilder.cs]: Ah, that's nice, thanks

**Review by premun — APPROVED** [review]
Seems to me that the limit is an unrelated problem that should not really affect this PR:
- This rather relies on the number of packages so keeping shorter hash will not prevent hitting the limit.
- The links are useless if they don't work?

Please log another issue around the limit to investigate what happens when we hit it and what to do about it.

**Comment by dkurepa** [comment]
we already have a piece of code that checks if the PR description is over 4k characters, so we don't need to worry if it's to long in this PR. Merging

**Comment by MilenaHristova** [comment]
Rolled out with https://github.com/dotnet/arcade-services/issues/3152

**Pull request merged**